applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_axisymmetric_element.hpp

```diff
@@ -0,0 +1,152 @@
+
+    Element::Pointer Create( IndexType NewId,
+                             GeometryType::Pointer pGeom,
+                             PropertiesType::Pointer pProperties ) const override;
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    // Turn back information as a string.
+    std::string Info() const override
+    {
+        std::stringstream buffer;
+        buffer << "U-Pw small strain different order axisymmetric Element #" << this->Id() << "\nConstitutive law: " << mConstitutiveLawVector[0]->Info();
+        return buffer.str();
+    }
+
+    // Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override
+    {
+        rOStream << "U-Pw small strain different order axisymmetric Element #" << this->Id() << "\nConstitutive law: " << mConstitutiveLawVector[0]->Info();
+    }
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+protected:
+
+    /// Member Variables
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    void CalculateBMatrix( Matrix &rB,
+                           const Matrix &GradNpT,
+                           const Vector &Np ) override;
+
+    double CalculateIntegrationCoefficient( const GeometryType::IntegrationPointsArrayType& IntegrationPoints,
+                                            const IndexType& PointNumber,
+                                            const double& detJ) override;
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+private:
+
+    /// Member Variables
+
+///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+    /// Serialization
+
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, Element )
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, Element )
+    }
+
+    // Assignment operator.
+    SmallStrainUPwDiffOrderAxisymmetricElement & operator=(SmallStrainUPwDiffOrderAxisymmetricElement const& rOther);
+
+    // Copy constructor.
+    SmallStrainUPwDiffOrderAxisymmetricElement(SmallStrainUPwDiffOrderAxisymmetricElement const& rOther);
+
+    // Private Operations
+
+    template < class TValueType >
+    inline void ThreadSafeNodeWrite(NodeType& rNode, const Variable<TValueType> &Var, const TValueType Value)
```

Comment:
just curious when do you need this?
if used when it is not needed it will significantly slow down the build

Reply:
I need this in the element to set average WATER_PRESSURE in intermediate nodes. It is called in function AssignPressureToIntermediateNodes.